hasher-matcher-actioner/src/OpenMediaMatch/blueprints/hashing.py

```diff
@@ -72,6 +77,33 @@ def is_valid_url(url: str) -> bool:
         return False
 
 
+def _get_and_check_content_length(
+    url: str, max_length: int = MAX_CONTENT_LENGTH
+) -> requests.Response:
+    """
+    Get URL content with explicit content length tracking.
+    Raises an exception if content length exceeds max_length.
+    """
+    # First check content length with HEAD request
+    head_resp = requests.head(url, timeout=30, allow_redirects=True)
+    head_resp.raise_for_status()
+
+    content_length = head_resp.headers.get("content-length")
+    if content_length is not None and int(content_length) > max_length:
+        abort(413, "Content too large")
+
+    # If content length is acceptable, proceed with GET request
+    response = requests.get(url, stream=True, timeout=30, allow_redirects=True)
+    response.raise_for_status()
+
+    # Double check content length from GET response
+    content_length = response.headers.get("content-length")
+    if content_length is not None and int(content_length) > max_length:
```

Comment:
Hmm, I think you can still get longer than the content-length if place you are calling doesn't return you a content-length header, which is the server discretion (it could also lie to you). I think if you want to truly defend against an attack, you need to keep track of the size yourself as you are streaming.... (see comment below)

Reply:
Yeah, usually you're better setting an explicit hard upper limit 200mb 500mb etc.